packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -521,6 +642,19 @@ assert(nullableString);
         break;
       }
     `,
+    `
+      const x = (t: string | null) => t;
```

Comment:
Two thoughts...
1. This appears to be the only added test case with a handler that is not a function literal. Maybe add a few more with varied return types?
2. I think this should be invalid, no? The handler returns `string | null`, and if we inline it, it does report.